### ISSUE TITLE
Object position/rotation editing system (gizmo)

### ISSUE THREAD

**Issue by jonlab** [jonlab/NewAtlantis#13]
Alex started to work on this.

**Comment by rolandcahen** [comment]
Alex, could you make the thickness of the gizmo a lille more skiny. Maybe the scaling should be only one sizing for the 3 dimension, otherwise it will be difficult to use, don't you think ?

**Comment by jonlab** [comment]
yes, I agree. I don't think that a non-homogeneous scaling is needed. Use case where you need this are very rare.
And maybe we would need a feedback of the values as well ?

**Comment by sachaamm** [comment]
Roland : actually the scaling is uniform.
Jonathan : about feedback values, I can do this but I think there are a lot of things more important to do for the January performance

**Comment by rolandcahen** [comment]
Hi Alex. 
Ergonomy question: 
At the moment, is is quite difficult to use the GIZMO to place NA_OBJECTS because when an OBJECT is selected in the SPACE window, the view changes. This is particularly true for big and composite OBJECTS in the middle of other big and composite OBJECTS. In addition it is difficult to navigate to a position while moving the object. Do you think this could be improved ?

**Comment by petesinc** [comment]
This seems to be reacting very slowly in the standalone I was unable to rotate an object on the vertical access... This is supposed to work in standalone?

**Comment by sachaamm** [comment]
Totally. Try to catch the right rotation axis , it is supposed to work

**Comment by benchang02** [comment]
some suggestions ( keep in mind I also might not know how to use the existing gizmo correctly)
- an "Edit" mode for the Viewer, like in Pd.  Click objects to select them, and HUD buttons + hotkeys for translate, rotate, scale as in Unity Editor and in Maya (Q=select, W=translate, E=rotate, R=scale, or make it configurable for different keyboards)
- since we often are moving around things that are really large, a different kind of camera view may be useful like the one in Unity or a 3D modeling app
- grid snap and rotation degree snap
- ability to toggle between local / global frame of reference
- is it possible to draw the gizmo as an overlay instead of regular scene object, so that it would maintain consistent size while being moved around?
- the translate tool was behaving oddly for me: clicking and dragging on, for example, the Z axis arrow would move the object along a different axis instead of the axis the arrow is pointing on
